app/components/find/courses/contact_details_component/view.rb

```diff
@@ -12,6 +12,30 @@ def initialize(course)
           super
           @course = course
         end
+
+        def show_contact_form_instead_of_email?
+          course.provider.provider_code.in?(course_information_config(:contact_forms).keys.map(&:to_s))
+        end
+
+        def show_address?
+          course.provider_code.in?(course_information_config(:show_address, :only_provider_codes)) && course.course_code.in?(course_information_config(:show_address, :only_course_codes))
+        end
+
+        def contact_form
+          contact_forms[course.provider.provider_code]
+        end
+
+        private
+
+        def contact_forms
+          course_information_config(:contact_forms).stringify_keys
+        end
+
+        def course_information_config(*path)
+          @course_information_config ||= Rails.application.config_for(:course_information)
```

Comment:
You are memoizing a `config_for` call? 🤔

Reply:
The `config_for` loads data from disk on every call.

Reply:
If it gets called often, why not add it to application config?
```ruby
config.x.course_information = config_for(:course_information)
```

and

```ruby
def course_information_config(*path)
  Rails.application.config.x.course_information.dig(*path)
end
```

Reply:
I ran some benchmarks and decided it isn't worth memoizing after all